Microsoft.Toolkit.Uwp.SampleApp/Controls/SampleAppMarkdownRenderer.cs

```diff
@@ -0,0 +1,302 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Toolkit.Parsers.Markdown;
+using Microsoft.Toolkit.Parsers.Markdown.Blocks;
+using Microsoft.Toolkit.Parsers.Markdown.Inlines;
+using Microsoft.Toolkit.Parsers.Markdown.Render;
+using Microsoft.Toolkit.Uwp.UI.Controls.Markdown.Render;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.UI;
+using Windows.UI.Text;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media;
+
+namespace Microsoft.Toolkit.Uwp.SampleApp.Controls
+{
+    internal class SampleAppMarkdownRenderer : MarkdownRenderer
+    {
+        public SampleAppMarkdownRenderer(MarkdownDocument document, ILinkRegister linkRegister, IImageResolver imageResolver, ICodeBlockResolver codeBlockResolver)
+            : base(document, linkRegister, imageResolver, codeBlockResolver)
+        {
+        }
+
+        /// <summary>
+        /// Adds extra adornments to Code Block, if it contains a Language.
+        /// </summary>
+        /// <param name="element">CodeBlock Element</param>
+        /// <param name="context">Render Context</param>
+        protected override void RenderCode(CodeBlock element, IRenderContext context)
+        {
+            // Renders the Code Block in the standard fashion.
+            base.RenderCode(element, context);
+
+            // Don't do any manipulations if the CodeLanguage isn't specified.
+            if (string.IsNullOrWhiteSpace(element.CodeLanguage))
+            {
+                return;
+            }
+
+            // Unify all Code Language headers for C#.
+            var language = element.CodeLanguage.ToUpper();
+            switch (language)
+            {
+                case "CSHARP":
+                case "CS":
+                    language = "C#";
```

Comment:
Should this not be in the parser somewhere vs the renderer?

Reply:
This is just for the header of the Sample App's Code Block that has a header specifying the language of the Code, just like docs.microsoft.com.